rerpc_test.go

```diff
@@ -307,6 +332,32 @@ func TestHandlerTwirp(t *testing.T) {
 			assert.Nil(t, json.Unmarshal(contents, got), "unmarshal JSON")
 			assert.Equal(t, got, expected, "unmarshaled Twirp status")
 		})
+
+		t.Run("bad_route", func(t *testing.T) {
+			r, err := http.NewRequest(
+				http.MethodPost,
+				fmt.Sprintf("%s/internal.ping.v1test.PingService/Foo", server.URL),
+				bytes.NewReader(nil),
+			)
+			assert.Nil(t, err, "create request")
+			r.Header.Set("Content-Type", rerpc.TypeProtoTwirp)
+
+			response, err := server.Client().Do(r)
+			assert.Nil(t, err, "make request")
+
+			testHeaders(t, response)
+			assert.Equal(t, response.StatusCode, http.StatusNotFound, "HTTP status code")
+
+			expected := &twirp.Status{
+				Code:    "bad_route",
+				Message: "no handler for path /internal.ping.v1test.PingService/Foo",
+			}
+			got := &twirp.Status{}
+			contents, err := io.ReadAll(response.Body)
+			assert.Nil(t, err, "read response body")
+			assert.Nil(t, json.Unmarshal(contents, got), "unmarshal JSON")
+			assert.Equal(t, got, expected, "unmarshaled Twirp status")
+		})
```

Comment:
These two blocks look identical to me - can we pull them out into a shared function? I know the tests I wrote aren't great about this right now :/

Reply:
Good idea. I can factor out a helper to test Twirp statuses, I think it will simplify a few other tests too.